### PR TITLE
Fixed to fetch all 4 fav-services

### PR DESCRIPTION
**Pull request by dihora-mayur**
- Setting default selected nav item to Open Positions if the navType being passed is "all"
- Updated logic to fetch "all" favorite data upon Favorites page load (this fix also addresses Tandem tabs not displayed if they are being favorite for the first time)

### CODE REVIEW

File: src/Containers/Favorites/Favorites.jsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export default connect(mapStateToProps, mapDispatchToProps)(
   withQueryParams({
     page: withDefault(NumberParam, 1),
     sortType: StringParam,
-    navType: withDefault(StringParam, 'open'),
+    navType: withDefault(StringParam, 'all'),
```
Comment (elizabeth-jimenez) [inline src/Containers/Favorites/Favorites.jsx:153]: 🦜

**Comment by elizabeth-jimenez** [comment]
Now that we are calling all 4 on initial, we don't need to call the individual ones again when we click on their tabs

**Comment by elizabeth-jimenez** [comment]
Successfully seeing the Tandem bug fixed

**Comment by dihora-mayur** [comment]
>

**Comment by dihora-mayur** [comment]
> Successfully seeing the Tandem bug fixed



> Now that we are calling all 4 on initial, we don't need to call the individual ones again when we click on their tabs

Good call, but it may still be a good idea to fetch individual data when switching between nav items. One instance i can think where this is necessary is when a person has multiple browser tabs open. Although refreshing the entire page does address this, just one instance i'm flagging. Not sure if there will be other cases.

**Comment by elizabeth-jimenez** [comment]
> > Successfully seeing the Tandem bug fixed
> 
> > Now that we are calling all 4 on initial, we don't need to call the individual ones again when we click on their tabs
> 
> Good call, but it may still be a good idea to fetch individual data when switching between nav items. One instance i can think where this is necessary is when a person has multiple browser tabs open. Although refreshing the entire page does address this, just one instance i'm flagging. Not sure if there will be other cases.

Yea, that's a good point; just unsure if we should make it work better for that edge case of multiple browser tabs open, as opposed to the more likely scenario of a single browser tab?

**Comment by elizabeth-jimenez** [comment]
> > > Successfully seeing the Tandem bug fixed
> > 
> > 
> > > Now that we are calling all 4 on initial, we don't need to call the individual ones again when we click on their tabs
> > 
> > 
> > Good call, but it may still be a good idea to fetch individual data when switching between nav items. One instance i can think where this is necessary is when a person has multiple browser tabs open. Although refreshing the entire page does address this, just one instance i'm flagging. Not sure if there will be other cases.
> 
> Yea, that's a good point; just unsure if we should make it work better for that edge case of multiple browser tabs open, as opposed to the more likely scenario of a single browser tab?

Edge case this wouldn't work for:
Go to page 2 of tab 'A'.
Go to a different tab.
Go back to tab 'A' - if we don’t call new results we will still have the results from page 2, which would require us to track the query per navtype (to either keep the same results and highlight the right pagination page, or reset to page 1 and trigger a call)

Mayur wins this battle 😈

**Comment by elizabeth-jimenez** [comment]
From Friday discussion:

As the code is now, before you click a navtab, all four call will fire off. So if you go to Favorites and start deleting **Available/Open** Favorites, all four calls will be made. I think same for pagination.

File: src/Components/FavoritePositions/FavoritePositions.jsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const TYPE_PV_TANDEM = 'pvTandem';
 const TYPE_OPEN_TANDEM = 'openTandem';
 
 const FavoritePositions = props => {
-  const [selected, setSelected] = useState(props.navType || TYPE_OPEN);
+  const [selected, setSelected] = useState(props.navType === 'all' ? TYPE_OPEN : props.navType);
```
Comment (elizabeth-jimenez) [inline src/Components/FavoritePositions/FavoritePositions.jsx:23]: From Friday Discussion - proposed code change:

Change this back to what it was.

**Comment by elizabeth-jimenez** [comment]
From Friday Discussion - proposed code change:
in `src/Containers/Favorites/Favorites.jsx`

```
useEffect(() => {
  const isNavAll = navType === 'all';

  if (!called ||
    !isEqual(userProfileFavoritePositionIsLoading, prevUserProfileFavoritePositionIsLoading)) {
    if (isNavAll) {
      // on initial page call, grab all favorites
      setCalled(true);
      nav$ = 'open';
      setQuery({ navType: 'open', page: 1 });
      props.bidListFetchData();
      getFavorites('pv');
      getFavorites('pvTandem');
      getFavorites('openTandem');
    } else {
      // Only fetch all if counts doesn't exist; otherwise fetch selected navType
      let type = 'all';
      if (keys(favoritePositions.counts).length) {
        type = navType;
      }
      setCalled(true);
      props.bidListFetchData();
      getFavorites(type);
    }
  }
});
```

**Review by elizabeth-jimenez — APPROVED** [review]
😸

**Pull request merged**